protocol_test.go

```diff
@@ -0,0 +1,65 @@
+package tarantool
```

Comment:
It is advisable not to test the internal implementation while possible. In this case, we can only test the public API (perhaps by adding useful methods to it).

Reply:
Well, I don't quite get how unit testing for internal implementation is a bad approach. Isn't it a basic design feature?

Reply:
It is a very debatable question:

1. https://medium.com/@thrawn01/why-you-should-never-test-private-methods-f822358e010
2. https://roadfiresoftware.com/2018/05/can-i-unit-test-private-methods/

To put it very briefly: it complicates the modification of the implementation.

Yes, it can be a different approach, but in this repository used a simple rules:

1. We are testing only the public interface.
2. If you really want to test integration with something (SSL, msgpack).
3. If you don't want to copy-paste or duplicate too much and it is very easy (better to avoid).

You could see such reasons in `export_test.go` files.

Reply:
Well, since these unit tests now not cover anything exceptional, removed